dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/QuayImageRegistry.java

```diff
@@ -203,6 +216,29 @@ private void insertQuayLastModifiedIntoLastBuilt(QuayTag quayTag, Tag tag) {
         return toolList;
     }
 
+    public Tool getToolFromNamespaceAndRepo(String namespace, String repository) {
+        try {
+            String name = namespace + "/" + repository;
+            QuayRepo repo = repositoryApi.getRepo(name, true);
+
+            Tool tool = new Tool();
+            // interesting, this relies upon our container object having the same fields
+            // as quay.io's repositories
+
+            // PLEASE NOTE : is_public is from quay.  It has NO connection to our is_published!
+            tool.setName(repo.getName());
+            tool.setNamespace(repo.getNamespace());
+            // tag all of these with where they came from
+            tool.setRegistry(Registry.QUAY_IO.toString());
```

Comment:
You'll want to use `getDockerPath()` instead now. `toString()` will give `QUAY_IO` after my changes.